src/redis_cmd.cc

```diff
@@ -1453,6 +1453,45 @@ class CommandHGetAll : public Commander {
   }
 };
 
+class CommandHRange : public Commander {
+ public:
+  Status Parse(const std::vector<std::string> &args) override {
+    if (args.size() != 6 && args.size() != 4) {
+      return Status(Status::RedisParseErr, errWrongNumOfArguments);
+    }
+    if (args.size() == 6 && Util::ToLower(args[4]) != "limit") {
+      return Status(Status::RedisInvalidCmd, errInvalidSyntax);
+    }
+    if (args.size() == 6) {
+      auto parseResult = ParseInt<int>(args_[5], /* base= */ 10);
+      if (!parseResult.IsOK())return Status(Status::RedisParseErr, errValueNotInterger);
+      limit_ = parseResult.GetValue();
+    }
+    return Commander::Parse(args);
+  }
+  Status Execute(Server *svr, Connection *conn, std::string *output) override {
+    if (args_.size() == 6 && limit_ < 0) {
+      *output = "*" + std::to_string(0) + CRLF;
+      return Status::OK();
+    }
+    Redis::Hash hash_db(svr->storage_, conn->GetNamespace());
+    std::vector<FieldValue> field_values;
+    rocksdb::Status s = hash_db.Range(args_[1], args_[2], args_[3], limit_, &field_values);
+    if (!s.ok()) {
+      return Status(Status::RedisExecErr, s.ToString());
+    }
+    *output = "*" + std::to_string(field_values.size() * 2) + CRLF;
+    for (const auto &fv : field_values) {
+      *output += Redis::BulkString(fv.field);
+      *output += Redis::BulkString(fv.value);
+    }
+    return Status::OK();
```

Comment:
Can we use `Redis::MultiBulkString`?

Reply:
> Can we use `Redis::MultiBulkString`?

Yes, But it need convert  `std::vector<FieldValue> field_values` to `std::vector<string> field_values`. Is there an elegant iterator conversion method in c++?

Reply:
I personally don't think it's very elegant to assemble RESP here. Maybe we could wrap it as a function that can passes `FieldValue`, or we could assemble it as `vector<string>` and pass it to our existing `Redis::MultiBulkString`.

Reply:
> I personally don't think it's very elegant to assemble RESP here. Maybe we could wrap it as a function that can passes `FieldValue`, or we could assemble it as `vector<string>` and pass it to our existing `Redis::MultiBulkString`

Yes. There are several places in the code where this direct assembly method is used. I can add a `Redis::BulkFieldValue` to replace them.